test/api/handlers_test.go

```diff
@@ -302,6 +304,70 @@ func testUserPUT(t *testing.T, at *test.AccountsTester) {
 	}
 }
 
+// testUserDELETE tests the DELETE /user endpoint.
+func testUserDELETE(t *testing.T, at *test.AccountsTester) {
+	u, c, err := test.CreateUserAndLogin(at, t.Name())
+	if err != nil {
+		t.Fatal("Failed to create a user and log in:", err)
+	}
+	// Delete the user.
+	at.Cookie = c
+	defer func() { at.Cookie = nil }()
+	r, _, err := at.Delete("/user", nil)
+	if err != nil || r.StatusCode != http.StatusNoContent {
+		t.Fatalf("Expected %d success, got %d '%s'", http.StatusNoContent, r.StatusCode, err)
+	}
+	// Create the user again.
+	u, c, err = test.CreateUserAndLogin(at, t.Name())
+	if err != nil {
+		t.Fatal("Failed to create a user and log in:", err)
+	}
+	// Create some data for this user.
+	sl, _, err := test.CreateTestUpload(at.Ctx, at.DB, u.User, 128)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = at.DB.DownloadCreate(at.Ctx, *u.User, *sl, 128)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = at.DB.RegistryWriteCreate(at.Ctx, *u.User)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = at.DB.RegistryReadCreate(at.Ctx, *u.User)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Try to delete the user without a cookie.
+	at.Cookie = nil
+	r, _, _ = at.Delete("/user", nil)
+	if r.StatusCode != http.StatusUnauthorized {
+		t.Fatalf("Expected %d, got %d", http.StatusUnauthorized, r.StatusCode)
+	}
+	// Delete the user.
+	at.Cookie = c
+	defer func() { at.Cookie = nil }()
+	r, _, err = at.Delete("/user", nil)
+	if err != nil || r.StatusCode != http.StatusNoContent {
+		t.Fatalf("Expected %d success, got %d '%s'", http.StatusNoContent, r.StatusCode, err)
+	}
+	// Make sure that the data is gone.
+	stats, err := at.DB.UserStats(at.Ctx, *u.User)
+	if err != nil {
```

Comment:
Shouldn't this return error not found?

Reply:
It's DB-level method, not a handler-level method. It's not its job to check that.